### PR TITLE
Multiplelookups 348

### PR DESCRIPTION
**Pull request by hudajkhan**
Addresses #348 .
Updates to resource template form enable passing back the array of lookup configurations (and not just one) so more than one lookup can be supported.
Options for menu now include an array of objects where each object includes the label/uri for the lookup as well as a "body" element that is the array of the results for that lookup.
Promises.all used to generate a promise per lookup requested, with catch blocks attached to each promise so that if one fails, the results of the others will still be passed along.
In the case where more than one lookup is used, titles are attached to the dropdown (using Menu.Header) from the lookup configuration. 
renderMenu is overridden customized for the typeahead component.
When a lookup returns no results or if there is an error that occurs, a message is appended under the appropriate heading (if more than one lookup source) or just directly (if only one lookup source).
Custom styling added to menu.css for dropdown header and some custom classes for error/empty result situations. 
Added one very basic test that just checks multiple values are available in state on search once saved there.  Would appreciate more input regarding tests!

### CODE REVIEW

**Comment by hudajkhan** [comment]
Added some more tests

File: src/styles/main.css [hunk 1]
```diff
@@ -171,3 +171,13 @@ input.rbt-input-main {
 .editor-navtabs a {
     color: #2F2424;
 }
+
+/** For lookup menu headers **/
+.dropdown-header {
+	font-weight:800;
+	color:#333;
```
Comment (jgreben) [inline src/styles/main.css]: I think we should add a background color, perhaps `background-color: lightgray;`?

**Comment by jgreben** [comment]
Looks good to me! I just think we need a background color to the multiple results header to make it pop a bit more. Also, it wouldn't hurt to add a second lookup ("urn:ld4p:qa:subjects:person") to the MonographInstance spoof file in order to have something readily at hand to see the functionality in action (for demo purposes):

File: src/styles/main.css (hunk 1 above)
Comment (hudajkhan) [reply src/styles/main.css]: Added background-color #d3d3d3 and updated the spoof file.

**Comment by hudajkhan** [comment]
I "resolved" the conversation.  I don't think I've ever done that before. Cool : )

File: __tests__/components/editor/InputLookupQA.test.js [hunk 2]
```diff
@@ -82,4 +114,38 @@ describe('<InputLookup />', () => {
...
+	   expect(multipleWrapper.state().options[1]).toEqual(multipleResults[1]) 
+
+  })
+  //Headers expected
+
+  it('should show menu headers with lookup source labels and values in the dropdown when provided results', () => {
+	  console.log("Wrapper instance");
+	  const instance = multipleWrapper.instance();
+	  const menuWrapper = shallow(instance.renderMenuFunc(multipleResults, p2Props));
+	  //console.log(menuWrapper.childAt(0).render().html());
```
Comment (jermnelson) [inline __tests__/components/editor/InputLookupQA.test.js]: I would remove these commented-out log statements.

**Comment by jermnelson** [comment]
Hi @hudajkhan, I made one small comment; looks good to me!

File: __tests__/components/editor/InputLookupQA.test.js (hunk 2 above)
Comment (hudajkhan) [reply __tests__/components/editor/InputLookupQA.test.js]: I've removed the commented out statements as well the first console.log which was just there for debugging the test.

**Pull request merged**